### PR TITLE
Fix/delta framework

### PR DESCRIPTION
**Pull request by Polymerase3**
### **User description**
Merge for the issue https://github.com/Polymerase3/phiper/issues/9

It is a pretty big PR:
- I dealt pretty much with everything that was described in the issue, so right now the code is clean, tested with coverage >90% and reproducible. 
- I explicitly tested the reproducibility by checking whether the results of two runs with the same seed are exactly the same
- I documented the function pretty extensively,
- I deleted the files with dead code,

One thing from this PR we should discuss with Alex: in our last discussion he recommended transforming the p-values to Z-scores with the sign of the Stouffer's T statistic. I did it, but I discovered, that the resolution is actually pretty small. With B = 150 000 permutations we are able to detect Z-scores in the range between (-4.5; 4.5), which is quite low. In the phipseq data we often detect much stronger signals and i/ve already performed some analysis with this framework, where i hit the ceiling.

It was a pretty complex and tricky task, but i figured out i can compute the standard deviation of the Stouffers T under the null from the permutations (commit 9bef10c777a70e00ffbd664120b1d6c0bb961b00). Then for each taxon i compute the standardized Stouffers T by dividing the raw T through the SD from the null. In this way i obtain Z-like scores, which i can then easily compare between the taxa and filter out the most interesting results.

___

### **PR Type**
Enhancement, Tests, Documentation

___

### **Description**
- **Major refactor of prevalence shift DELTA framework** with bitset-based implementation for improved efficiency and accuracy

- **Implemented standardized T-statistics and Z-scores** by computing null standard deviation from permutations, enabling better signal detection across taxa

- **Added paired design support** via new `paired_by` parameter with automatic detection of paired vs unpaired contrasts

- **Introduced fold-change and cross-prevalence summaries** with multiple aggregation modes (sum, mean, max, median)

- **Refactored parallelization** to use `future` backend with dynamic scheduling and thread limiting

- **Replaced mean delta outputs with max delta** (maximum absolute peptide-level difference) for better effect size representation

- **Comprehensive test coverage** with 1032 lines of tests for main function and 463 lines for visualization functions

- **Renamed visualization functions** for consistency: `deltaplot_prevalence` → `deltaplot`, `ecdf_prevalence` → `ecdf_plot`, `forest_delta` → `forestplot`

- **Added new output columns**: `T_obs_stand` (standardized T), `Z_from_p` (signed Z from p-value), `b` (actual permutations used)

- **Extended documentation** with mathematical details, LaTeX equations, and comprehensive parameter descriptions

- **Added interactive visualization functions** with plotly support for forest plots and ECDF plots

- **Cleaned up codebase** by removing dead code files and improving code organization

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Raw Peptide Data"] -->|"Bitset Representation"| B["Efficient Data Processing"]
  B -->|"Permutation Loop"| C["Compute Stouffer T & Null SD"]
  C -->|"Standardization"| D["T_obs_stand & Z_from_p"]
  D -->|"Visualization"| E["deltaplot, forestplot, ecdf_plot"]
  B -->|"Paired/Unpaired Design"| F["Contrast Detection"]
  F -->|"Fold-change & Cross-prev"| G["Summary Statistics"]
  G -->|"Output"| H["Enhanced Results Table"]
```

<details><summary><h3>File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>shift_computing.cpp</strong><dd><code>Add null standard deviation computation for Stouffer T statistic</code></dd></summary>
<hr>

src/shift_computing.cpp

<ul><li>Added <code>#include <limits></code> header for numeric limits support<br> <li> Replaced <code>mean_delta</code> and <code>mean_delta_w</code> outputs with <code>max_delta</code> (maximum <br>absolute peptide-level difference)<br> <li> Added <code>T_null_sd</code> output field to return standard deviation of <br>Stouffer's T statistic under the null hypothesis from permutations<br> <li> Implemented computation of <code>sum_T</code> and <code>sum_T2</code> during permutation loop to <br>calculate null standard deviation using variance formula<br> <li> Updated return lists in both paired and unpaired design branches to <br>include new <code>T_null_sd</code> field and remove weighted mean delta field</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-b223083e0d8d07a9d8dd429578a8b7b9a003c0228657110397eb0117695223ae">+57/-26</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>prevalence-DELTA_test.R</strong><dd><code>Major refactor: bitset-based implementation with paired design support</code></dd></summary>
<hr>

R/prevalence-DELTA_test.R

<ul><li>Completely refactored function signature: removed <code>seed</code>, <code>parallel</code>, <br><code>auto_fetch_library</code>, <code>stream_path</code>, <code>return_results</code>, <code>append_stream</code> <br>parameters; added <code>paired_by</code>, <code>fold_change</code>, <code>cross_prev</code> parameters<br> <li> Rewrote documentation with extensive mathematical details for all 7 <br>steps of the procedure, including LaTeX equations for smoothing, <br>z-scores, weights, Stouffer combination, permutation scheme, and <br>multiplicity adjustment<br> <li> Replaced data processing pipeline: now uses bitset-based <br>representation for efficient peptide presence data and direct C++ <br>calls instead of R-based permutation logic<br> <li> Implemented strict input validation guard ensuring at most one <br>positive per (subject_id, peptide_id, group_col, group_value) <br>combination<br> <li> Added support for paired designs via <code>paired_by</code> parameter with <br>automatic detection of paired vs unpaired contrasts<br> <li> Introduced optional fold-change and cross-prevalence summaries with <br>multiple aggregation modes (sum, mean, max, median)<br> <li> Refactored parallelization to use <code>future</code> backend with dynamic <br>scheduling and thread limiting to prevent oversubscription<br> <li> Added new output columns: <code>T_obs_stand</code> (standardized T), <code>Z_from_p</code> <br>(signed Z from p-value), <code>b</code> (actual permutations used)<br> <li> Replaced <code>mean_delta</code> and <code>mean_delta_w</code> with <code>max_delta</code> in output<br> <li> Implemented file-based progress tracking with locking mechanism for <br>thread-safe updates</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-4265218b02726a020937c607a5ba893074a21fe31fdd83ca97c4f2f57c61efd6">+996/-561</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>prevalence-POP-test.R</strong><dd><code>Simplify SQL table column extraction using base R</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

R/prevalence-POP-test.R

<ul><li>Replaced <code>dplyr::slice_head(lib_handle, n = 0)</code> with <code>head(lib_handle, 0)</code> <br>in four locations (lines 619, 1082, 1106, 1382)<br> <li> Simplifies code by using base R <code>head()</code> function instead of dplyr <br>equivalent for getting column names from SQL tables</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-7321a6ef649b4acb13729f597810ed7edf5ffb3d9f83c31fdf5e0ede2fbffddc">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>NAMESPACE</strong><dd><code>Rename visualization functions for consistency</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

NAMESPACE

<ul><li>Renamed exported functions: <code>deltaplot_prevalence</code> → <code>deltaplot</code>, <br><code>deltaplot_prevalence_interactive</code> → <code>deltaplot_interactive</code><br> <li> Renamed exported functions: <code>ecdf_prevalence</code> → <code>ecdf_plot</code>, <br><code>ecdf_prevalence_interactive</code> → <code>ecdf_plot_interactive</code><br> <li> Renamed exported functions: <code>forest_delta</code> → <code>forestplot</code>, <br><code>forest_delta_plotly</code> → <code>forestplot_interactive</code></ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-b6b6854a02ae177f8860f49654ff250c1554d021ae434b6efdbe99d00bdc6055">+6/-11</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>scatter_interactive.Rd</strong><dd><code>Extend scatter_interactive with background and styling options</code></dd></summary>
<hr>

man/scatter_interactive.Rd

<ul><li>Added new parameters to <code>scatter_interactive()</code>: <code>peplib</code>, <br><code>category_colors</code>, <code>show_background</code>, <code>background_df</code>, <code>background_name</code>, <br><code>background_color</code>, <code>background_size</code>, <code>background_alpha</code>, <code>background_max_n</code>, <br><code>background_seed</code><br> <li> Added <code>point_line_width</code> and <code>point_line_color</code> parameters for point <br>styling<br> <li> Extends functionality to support background peptide visualization and <br>category-based coloring</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-32713d4ea632d834952eb61567ecd00334df15fde2956f1f6191911e788add22">+13/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>ph_prevalence_shift.Rd</strong><dd><code>Comprehensive documentation update with mathematical details</code></dd></summary>
<hr>

man/ph_prevalence_shift.Rd

<ul><li>Updated title capitalization and description to reflect antigen-wide <br>(not genome-wide) scope<br> <li> Expanded parameter documentation with mathematical notation and <br>detailed explanations for all statistical modes<br> <li> Added new parameters: <code>paired_by</code>, <code>fold_change</code>, <code>cross_prev</code>; removed: <br><code>seed</code>, <code>parallel</code>, <code>auto_fetch_library</code>, <code>stream_path</code>, <code>return_results</code>, <br><code>append_stream</code><br> <li> Reorganized Details section into 7 numbered steps with comprehensive <br>LaTeX equations for each statistical procedure<br> <li> Added new "Input requirements" and "Peptide library resolution" <br>subsections with fallback logic explanation<br> <li> Added "Parallelization" section documenting <code>future</code> backend usage and <br>thread management<br> <li> Expanded return value documentation with detailed descriptions of all <br>output columns including new <code>T_obs_stand</code>, <code>Z_from_p</code>, <code>b</code>, and optional <br>fold-change/cross-prevalence columns</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-376c3227d9a8ac49035b644775c51d19e821860f21a596e0c3b9127d0d630791">+199/-77</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>forestplot_interactive.Rd</strong><dd><code>New interactive forest plot visualization function documentation</code></dd></summary>
<hr>

man/forestplot_interactive.Rd

<ul><li>New file documenting <code>forestplot_interactive()</code> function for creating <br>interactive forest plots<br> <li> Provides plotly-based visualization of top/bottom features ranked by <br>Stouffer statistics or standardized variants<br> <li> Supports filtering by significance level and multiple statistic types <br>(raw T, standardized T, Z from p-value)<br> <li> Includes customizable styling options for arrows, labels, colors, and <br>fonts<br> <li> Contains working example demonstrating usage with simulated results <br>table</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-b91f1f0e650bb442428991f91d0dae7cc40d8c07195eae439441052e19434435">+149/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>forestplot.Rd</strong><dd><code>Documentation for forestplot function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

man/forestplot.Rd

<ul><li>New documentation file for <code>forestplot()</code> function showing forest plot <br>of top/bottom Stouffer T statistics by rank<br> <li> Documents parameters for filtering, coloring, sizing, and <br>customization of forest plot appearance<br> <li> Includes example showing how to create forest plots with different <br>statistics (<code>T</code>, <code>T_stand</code>, <code>Z_from_p</code>)</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-24fc832660bd2d881b0b4ab96b50f3dccbb3c67f1e576fb253d1a89426c45d56">+164/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>deltaplot_interactive.Rd</strong><dd><code>Documentation for interactive deltaplot function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

man/deltaplot_interactive.Rd

<ul><li>New documentation for interactive plotly version of delta prevalence <br>plot<br> <li> Documents parameters for point styling, smoothing, arrows, and labels<br> <li> Includes example workflow using <code>ph_prevalence_compare()</code> output</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-d268204e2bf7a2f1a99145ed0b26b2002004ece7772fd6e2c492ee5beddce453">+142/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>deltaplot.Rd</strong><dd><code>Documentation for static deltaplot function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

man/deltaplot.Rd

<ul><li>New documentation for static ggplot delta prevalence visualization<br> <li> Documents parameters for jitter, smoothing, arrows, and axis labels<br> <li> Includes complete example showing prevalence shift analysis workflow</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-f90970aa458c9c87aa9377849a4317b3164bfdc54a9eda11d3d4de6f63d0b523">+140/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>ecdf_plot.Rd</strong><dd><code>Documentation for ECDF prevalence plot function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

man/ecdf_plot.Rd

<ul><li>New documentation for ECDF plot of per-peptide prevalences for two <br>groups<br> <li> Documents parameters for line styling, median lines, and KS test <br>annotation<br> <li> Includes example demonstrating ECDF comparison between groups</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-5cb357c56852753f2b9fea02dd26a239b8490842ec497ae49c03706e02affadc">+119/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>ecdf_plot_interactive.Rd</strong><dd><code>Documentation for interactive ECDF plot function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

man/ecdf_plot_interactive.Rd

<ul><li>New documentation for interactive plotly ECDF prevalence plot<br> <li> Documents parameters for line styling, median lines, and KS test <br>options<br> <li> Includes example workflow for interactive ECDF visualization</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-1df353f368992b18effe016e9258f4ad0016259e92eb0c69bf0fe92298f27ae4">+113/-0</a>&nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>legacy-backends.R</strong><dd><code>Improve roxygen2 documentation formatting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

R/legacy-backends.R

<ul><li>Updated roxygen2 documentation tags for memory-backed <code>phip_data</code> object <br>builder function<br> <li> Changed from inline comments to formal <code>@title</code> and <code>@description</code> tags <br>for better documentation generation</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-6e8048122c47e5deb773c31d71458df88120f08737290d93498f484777ab7358">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test-prevalence-DELTA_test.R</strong><dd><code>Comprehensive test suite for prevalence shift DELTA framework</code></dd></summary>
<hr>

tests/testthat/test-prevalence-DELTA_test.R

<ul><li>Added comprehensive test suite (1032 lines) for <code>ph_prevalence_shift()</code> <br>function covering unpaired/paired designs, error handling, and <br>reproducibility<br> <li> Tests validate standardized T statistics (<code>T_obs_stand</code>), Z-scores from <br>p-values (<code>Z_from_p</code>), and permutation p-values<br> <li> Includes tests for fold_change and cross_prev summary modes, weight <br>modes (se_invvar), prevalence stratification, and RNG reproducibility<br> <li> Tests verify proper handling of edge cases like zero variance, missing <br>columns, and bitset dimension mismatches</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-5882413469747f46ff27aee1532d5a2fcf2976561c88ac640eecf9d5da8f22b1">+1032/-0</a></td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test-prevalence-DELTA_plots.R</strong><dd><code>Test suite for prevalence DELTA visualization functions</code>&nbsp; &nbsp; </dd></summary>
<hr>

tests/testthat/test-prevalence-DELTA_plots.R

<ul><li>Added 463-line test suite for visualization functions: <code>deltaplot()</code>, <br><code>deltaplot_interactive()</code>, <code>forestplot()</code>, <code>forestplot_interactive()</code>, <br><code>ecdf_plot()</code>, and <code>ecdf_plot_interactive()</code><br> <li> Tests validate plot reproducibility with seeds, input validation, <br>filtering options, and interactive plotly output<br> <li> Includes tests for helper functions like <code>.ph_forestplot_prepare()</code> and <br><code>.empty_placeholder_plot()</code><br> <li> Verifies color scaling, median lines, KS test annotations, and smooth <br>curve generation</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-1b9d0f8a51c2693e72c19e94206d2ff56d68542c04b0be7f8b6b608547e87162">+463/-0</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Dependencies</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>DESCRIPTION</strong><dd><code>Add visualization dependencies and fix formatting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

DESCRIPTION

<ul><li>Added <code>vdiffr</code> to Suggests for visual regression testing of plots<br> <li> Added <code>plotly</code> to Imports for interactive visualization support<br> <li> Fixed trailing whitespace in <code>Rcpp</code> dependency line</ul>

</details>

  </td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-9cc358405149db607ff830a16f0b4b21f7366e3c99ec00d52800acebe21b231c">+4/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Additional files</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td><strong>RcppExports.R</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-fe103c2e1ca566e20e16d95e118a48e77eac3a0ae97a3a0c89d3905c71ca017c">+0/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>prev_shift_helpers.R</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-b65587aed72048e1b39cebfd4b8eaf3b006a6a46c4b2bdb1f9b2bb67d78fb07e">+0/-640</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>prevalence-DELTA_plots.R</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-3596fa5e1e8e68b871f0dabded86542f842364b258e766722884eb7b7fa72ba5">+1869/-627</a></td>

</tr>

<tr>
  <td><strong>prevalence-DELTA_test2.R</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-52976c6267ddbe713ed51a21d491a580dea2afe76475caa4110fca565f65120f">+0/-855</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>deltaplot_prevalence.Rd</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-399c04fcf3c2b062084aee2fa271106c743b15511bd68a211d79276331f7b161">+0/-74</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>forest_delta.Rd</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-d408db90f2918e1678b580d103a1fdf279e6eade5bed8c508198e3a9bf3645b6">+0/-96</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ph_prevalence_shift2.Rd</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-a6f79d7593884197a2362ce190d70dbc6fcf984737e6aafbabd49d1fc037b7ec">+0/-159</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>RcppExports.cpp</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-13dc68d7d985baff5c1cac41e1471ed9fe4db63a2bea80c2cdf20fd795541062">+0/-29</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>perm_bitset_parallel.cpp</strong></td>
  <td><a href="https://github.com/Polymerase3/phiper/pull/13/files#diff-910f1c8cdac69bc2caf856826d94acbfcfc062c2a9613eb7b69d43b9daaea39c">+0/-431</a>&nbsp; </td>

</tr>
</table></details></td></tr></tbody></table>

</details>

___

### CODE REVIEW

File: R/prevalence-DELTA_plots.R [hunk 1]
```diff
@@ -601,73 +722,129 @@ forest_delta_plotly <- function(
...
+  if (isTRUE(add_signed_z_from_p) || identical(stat, "Z")) {
+    if (!"p_perm" %in% names(x)) {
+      stop("Column `p_perm` is required to compute signed Z from p-values.")
+    }
+    p_two <- x$p_perm
+    p_two[!is.na(p_two)] <- pmin(
+      pmax(p_two[!is.na(p_two)], .Machine$double.eps),
+      1 - .Machine$double.eps
+    )
+    x$Z_signed_from_p_2s <- sign(x$T_obs) * stats::qnorm(1 - p_two / 2)
```
Comment (Alon-Alexander) [inline R/prevalence-DELTA_plots.R]: This is duplicated code. Maybe extract to small helper function (as we can see, this content changes ;)

File: R/prevalence-DELTA_plots.R [hunk 2]
```diff
@@ -601,73 +722,129 @@ forest_delta_plotly <- function(
...
   }
 
   df_rk <- dplyr::filter(x, .data$rank == rank_of_interest)
+
   if (!identical(filter_significant, "none")) {
-    if (!filter_significant %in% colnames(df_rk)) stop(paste0("column '", filter_significant, "' not found"))
+    if (!filter_significant %in% colnames(df_rk)) {
```
Comment (Alon-Alexander) [inline R/prevalence-DELTA_plots.R]:
```suggestion
    if (!(filter_significant %in% colnames(df_rk))) {
```

File: R/prevalence-DELTA_test.R [hunk 3]
```diff
@@ -1,701 +1,1136 @@
...
+#' **Input requirements.**
 #'
-#' **Notes.**
-#' - If a stratum doesn’t contain exactly two group levels, it’s skipped.
-#' - Progress can be printed every `progress_perm_every` permutations.
+#' - `exist_col` is treated as 0/1 presence.
+#' - There must be **at most one positive** per
+#'   (`subject_id`, `peptide_id`, `group_col`, `group_value`); paired designs
+#'   can have up to two positives across the two group levels. Violations
+#'   trigger an error.
```
Comment (Alon-Alexander) [inline R/prevalence-DELTA_test.R]: Can you give super short example here or elaborate more.
I think I get it, but would be great if it's completely obvious what you mean.

File: R/prevalence-DELTA_test.R [hunk 4]
```diff
@@ -1,701 +1,1136 @@
...
+          "progress=%s [C]",
+          if (is.na(prog)) "n/a" else sprintf("%.1f%%", prog)
+        ))
       }
     }
+
+    row
+  }
+
+  # --- 7/8) Computation path: sequential vs parallel --------------------------
```
Comment (Alon-Alexander) [inline R/prevalence-DELTA_test.R]: Note: I didn't read most of the function as it's very long and I don't fully understand the delta framework yet.
I trust you that the implementation works 😬

File: tests/testthat/test-prevalence-DELTA_test.R
Comment (Alon-Alexander) [inline tests/testthat/test-prevalence-DELTA_test.R]: Amazing file! 👏👏👏

**Review by Alon-Alexander — CHANGES_REQUESTED** [review]
This is great!
I did leave a lot of comments but this is a huge PR, so it makes sense (to me).

File: R/prevalence-DELTA_plots.R [hunk 5]
```diff
@@ -433,102 +433,185 @@ forest_delta <- function(
...
+      plot = .empty_placeholder_plot(
+        sprintf("No rows to plot (rank='%s')", rank_of_interest)
+      )
+    ))
+  }
+
+  # choose statistic used for ranking and selection
+  if (identical(stat, "Z")) {
+    if (!"Z_signed_from_p_2s" %in% names(df_rk)) {
+      stop("`Z_signed_from_p_2s` not found. Set `add_signed_z_from_p = TRUE` or pre-compute it.")
```
[the remark this answers is not in the document]
Comment (Polymerase3) [reply R/prevalence-DELTA_plots.R]: I actually changed this. I wanted to discuss it with Alex, but siince he is on vacation, we would need to postpone this a little bit. ph_prevalence_shift  computes now the raw Stouffer's T, the Z scores from the p-values and the standardized Stouffer's T, so the user can choose, which type of statistic is to be plotted. But we need to discuss, which one of those 3 is actually the best default for now.

File: R/prevalence-DELTA_plots.R [hunk 6]
```diff
@@ -433,102 +433,185 @@ forest_delta <- function(
...
+  .pt_to_gg <- function(pt) pt / 2.845
 
-  if (isTRUE(add_signed_z_from_p)) {
-    x$Z_signed_from_p_2s <- sign(x$T_obs) * stats::qnorm(1 - x$p_perm/2)
+  # compute signed Z from p if requested or needed for plotting
+  if (isTRUE(add_signed_z_from_p) || identical(stat, "Z")) {
+    if (!"p_perm" %in% names(x)) {
+      stop("Column `p_perm` is required to compute signed Z from p-values.")
+    }
+    p_two <- x$p_perm
```
[the remark this answers is not in the document]
Comment (Polymerase3) [reply R/prevalence-DELTA_plots.R]: Id leave it as it is. With `p_perm` I can track the origin of the p-value (permutations) and I know it’s the raw, uncorrected p. The one- vs two-sided detail isn’t crucial here. In research, tests are (and generally should be) two-sided most of the time anyway, unless you’re intentionally trying to p-hack the results.

File: R/prevalence-DELTA_plots.R [hunk 7]
```diff
@@ -601,73 +722,129 @@ forest_delta_plotly <- function(
...
         margin = list(l = 20, r = 20, t = 60, b = 20)
       )
     return(list(data = df_rk, plot = plt))
   }
 
-  n_pos <- sum(df_rk$T_obs > 0, na.rm = TRUE)
-  n_neg <- sum(df_rk$T_obs < 0, na.rm = TRUE)
-  top_pos <- df_rk |> dplyr::arrange(dplyr::desc(.data$T_obs)) |> dplyr::slice_head(n = min(n_pos_each, n_pos))
-  top_neg <- df_rk |> dplyr::arrange(.data$T_obs)                 |> dplyr::slice_head(n = min(n_neg_each, n_neg))
+  # choose statistic used for ranking and selection
```
[the remark this answers is not in the document]
Comment (Polymerase3) [reply R/prevalence-DELTA_plots.R]: I prepared an internal helper .ph_forestplot_prepare and documented it. It is now used both by the forestplot and forestplot_interactive.

File: R/prevalence-DELTA_test.R [hunk 8]
```diff
@@ -1,701 +1,1136 @@
...
-#' @param x A `phip_data` (recommended; must carry `peptide_library` when
-#'   `rank_cols` include non-`peptide_id`) or a data.frame with the necessary
-#'   columns.
+#' **Peptide library resolution.**
+#'
+#' When `rank_cols` includes non-`"peptide_id"` ranks, a peptide library is
+#' required. It is resolved in the following order:
+#'
+#' 1. `x$peptide_library` if `x` is a `phip_data` with an attached library.
+#' 2. The explicit `peptide_library` argument.
```
[the remark this answers is not in the document]
Comment (Polymerase3) [reply R/prevalence-DELTA_test.R]: Done!

File: R/prevalence-DELTA_test.R [hunk 9]
```diff
@@ -1,701 +1,1136 @@
...
+    dplyr::distinct(peptide_id) |>
+    dplyr::arrange(peptide_id) |>
+    dplyr::collect() |>
+    dplyr::pull(peptide_id)
+
+  # --- BUILD HITS DIRECTLY FROM DISTINCT POSITIVES ----------------------------
+  pos_pairs <- df_long |>
+    dplyr::filter(!!rlang::sym(exist_col) > 0L) |>
+    dplyr::distinct(subject_id, peptide_id) |>
+    dplyr::collect()
```
[the remark this answers is not in the document]
Comment (Polymerase3) [reply R/prevalence-DELTA_test.R]: Yes, usually the mean prevalence of hits across the whole dataset is around ~1-2%. At some point you need to collect it to the memory, and i figured out that a quite effective way to do this is to keep the position of the 1s in the dataset and then build a bitset out of this. There is prolly even a more effective way to do this and optimize this, but i used this function already on large datasets and it did not cause any major RAM problems.

File: R/prevalence-DELTA_test.R [hunk 10]
```diff
@@ -1,701 +1,1136 @@
...
+  on.exit(options(op_old), add = TRUE)
+
+  # ---- avoid inner oversubscription (TBB inside RcppParallel) ----
+  if (rlang::is_installed("RcppParallel")) {
+    RcppParallel::setThreadOptions(numThreads = 1) # fairness > peak single-task
+  } #            speed
+  Sys.setenv(
+    OMP_NUM_THREADS = "1", MKL_NUM_THREADS = "1",
+    OPENBLAS_NUM_THREADS = "1"
+  )
```
[the remark this answers is not in the document]
Comment (Polymerase3) [reply R/prevalence-DELTA_test.R]: You are right, i used withr::with_envvar to handle this, since withr is already in the dependencies.
